include/ttmlir/Scheduler/Scheduler.h

```diff
@@ -0,0 +1,44 @@
+// SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef TTMLIR_SCHEDULER_SCHEDULER_H
+#define TTMLIR_SCHEDULER_SCHEDULER_H
+
+#include <memory>
+
+#include "mlir/IR/Operation.h"
+
+namespace mlir::tt::scheduler {
+
+class Scheduler {
+public:
+  // Constructor taking an MLIR Operation (or a module)
+  Scheduler(mlir::Operation *root);
+
+  // Copy constructor
+  Scheduler(const Scheduler &scheduler);
+
+  // Method to get the next set of schedulable operations
+  llvm::SmallVector<mlir::Operation *> getScheduleableOps();
+
+  // Method to check if an operation can be scheduled
+  bool canSchedule(mlir::Operation *op);
+
+  // Method to schedule an operation
+  void schedule(mlir::Operation *op);
+
+  // Method to take a snapshot of the scheduler
+  std::unique_ptr<Scheduler> snapshot();
+
+private:
+  llvm::DenseSet<mlir::Operation *> scheduledOps;
```

Comment:
This will not give us ability to print final result schedule?

Reply:
Not sure what you mean by that? Do you want me to expose getter for scheduledOps?

Reply:
Getter for scheduled OPs in order as they are scheduled.